### PR TITLE
feat: Generate ARM template for Windows agent pool for different VHD images

### PR DESCRIPTION
**Pull request by AbelHu**
**Reason for Change**:

feat: Generate ARM template for Windows agent pool for different VHD images

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/virtualmachines.go [hunk 1]
```diff
@@ -498,7 +498,25 @@ func createAgentAvailabilitySetVM(cs *api.ContainerService, profile *api.AgentPo
 	storageProfile := compute.StorageProfile{}
 
 	if profile.IsWindows() {
-		storageProfile.ImageReference = createWindowsImageReference(profile.Name, cs.Properties.WindowsProfile)
```
Comment (mainred) [inline pkg/engine/virtualmachines.go:501]: Should we add these into `createWindowsImageReference` instead?

File: pkg/engine/virtualmachinescalesets.go [hunk 2]
```diff
@@ -665,7 +665,27 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 	vmssStorageProfile := compute.VirtualMachineScaleSetStorageProfile{}
 
 	if profile.IsWindows() {
-		vmssStorageProfile.ImageReference = createWindowsImageReference(profile.Name, cs.Properties.WindowsProfile)
+		if profile.HasImageRef() {
```
Comment (mainred) [inline pkg/engine/virtualmachinescalesets.go:668]: ditto

File: pkg/engine/virtualmachines.go (hunk 1 above)
Comment (AbelHu) [reply pkg/engine/virtualmachines.go:501]: I planned to remove createWindowsImageReference  later. I think that keeping the logic here as Linux does makes it more readable.

Comment (mainred) [reply pkg/engine/virtualmachines.go:501]: But it leaves `createWindowsImageReference` in a wrong logic, I prefer we leave a TODO comment to fix it later.  It depends on you.

**Review by mainred — APPROVED** [review]

**Pull request merged**